xdem/coreg.py

```diff
@@ -2155,134 +2180,233 @@ def warp_dem(
+) -> tuple[xdem.DEM, xdem.coreg.Coreg, pd.DataFrame, NDArrayf]:
     """
     A one-line function to coregister a selected DEM to a reference DEM.
-    Reads both DEMs, reprojects them on the same grid, mask content of shpfile, filter steep slopes and outliers, \
-run the coregistration, returns the coregistered DEM and some statistics.
+
+    Reads both DEMs, reprojects them on the same grid, mask pixels based on shapefile(s), filter steep slopes and \
+outliers, run the coregistration, returns the coregistered DEM and some statistics.
     Optionally, save the coregistered DEM to file and make a figure.
+    For details on masking options, see `create_inlier_mask` function.
 
-    :param src_dem_path: path to the input DEM to be coregistered
-    :param ref_dem: path to the reference DEM
+    :param src_dem_path: Path to the input DEM to be coregistered
+    :param ref_dem_path: Path to the reference DEM
     :param out_dem_path: Path where to save the coregistered DEM. If set to None (default), will not save to file.
-    :param shpfile: path to a vector file containing areas to be masked for coregistration
-    :param coreg_method: The xdem coregistration method, or pipeline. If set to None, DEMs will be resampled to \
-ref grid and optionally filtered, but not coregistered. Will be used in priority over hmode and vmode.
-    :param hmode: The method to be used for horizontally aligning the DEMs, e.g. Nuth & Kaab or ICP. Can be any \
-of {list(vmodes_dict.keys())}.
-    :param vmode: The method to be used for vertically aligning the DEMs, e.g. mean/median bias correction or \
-deramping. Can be any of {list(hmodes_dict.keys())}.
-    :param deramp_degree: The degree of the polynomial for deramping.
-    :param grid: the grid to be used during coregistration, set either to "ref" or "src".
-    :param filtering: if set to True, filtering will be applied prior to coregistration
-    :param plot: Set to True to plot a figure of elevation diff before/after coregistration
+    :param coreg_method: The xdem coregistration method, or pipeline.
+    :param grid: The grid to be used during coregistration, set either to "ref" or "src".
+    :param resample: If set to True, will reproject output Raster on the same grid as input. Otherwise, only \
+the array/transform will be updated (if possible) and no resampling is done. Useful to avoid spreading data gaps.
+    :param shp_list: A list of one or several paths to shapefiles to use for masking.
+    :param inout: A list of same size as shp_list. For each shapefile, set to 1 (resp. -1) to specify whether \
+to mask inside (resp. outside) of the polygons. Defaults to masking inside polygons for all shapefiles.
+    :param filtering: If set to True, filtering will be applied prior to coregistration.
+    :param dh_max: Remove pixels where abs(src - ref) is more than this value.
+    :param nmad_factor: Remove pixels where abs(src - ref) differ by nmad_factor * NMAD from the median.
```

Comment:
This is much better than in the mask generation docstring! (See my comment above)